src/dashboards/components/RefreshContext.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {createContext, useReducer, useCallback, FC} from 'react'
+import {useSelector, useDispatch} from 'react-redux'
+import moment from 'moment'
+
+// Types
+import {
+  CustomTimeRange,
+  AutoRefreshStatus,
+  AutoRefresh,
+  AppState,
+} from 'src/types'
+
+// Actions
+import {
+  setAutoRefreshInterval,
+  setAutoRefreshStatus,
+  setAutoRefreshDuration,
+  setInactivityTimeout,
+} from 'src/shared/actions/autoRefresh'
+
+export const AutoRefreshContext = createContext(null)
+
+interface AutoRefreshState {
+  duration: CustomTimeRange
+  inactivityTimeout: string
+  inactivityTimeoutCategory: string
+  refreshMilliseconds: AutoRefresh
+}
+
+const calculateTimeout = (timeout, timeoutUnit) => {
+  const timeoutNumber = parseInt(timeout)
+  const startTime = moment(new Date())
+  const copyStart = startTime.unix()
+  const endTime = startTime.add(timeoutNumber, timeoutUnit[0].toLowerCase())
+  const cutoff = endTime.unix() - copyStart
+  return cutoff * 1000
+}
+
+export const createAutoRefreshInitialState = (
+  override = {}
+): AutoRefreshState => {
+  return {
+    inactivityTimeout: 'None',
+    inactivityTimeoutCategory: 'Hours',
+    duration: {
+      lower: new Date().toISOString(),
+      upper: new Date().toISOString(),
+      type: 'custom',
+    } as CustomTimeRange,
```

Comment:
i've been bitten by using this instead of selectabletime range, or whatever the parent class is of this, when it came to saving and integrating into other parts of the app. bite the bullet and figure out the typescript error now, promise it's worth it

Reply:
No error here, just a piece of code that should have been removed once i made the interface